broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java

```diff
@@ -167,29 +167,19 @@ private void validateDataConfig(final DataCfg dataCfg) {
   }
 
   private void validateBackupCfg(final BackupStoreCfg backup) {
-    if (backup.getStore() == BackupStoreType.NONE) {
-      LOG.warn("No backup store is configured. Backups will not be taken");
-      return;
-    }
-
-    if (backup.getStore() == BackupStoreType.S3) {
-      final var s3Config = backup.getS3();
-      try {
-        final S3BackupConfig storeConfig =
-            new Builder()
-                .withBucketName(s3Config.getBucketName())
-                .withEndpoint(s3Config.getEndpoint())
-                .withRegion(s3Config.getRegion())
-                .withCredentials(s3Config.getAccessKey(), s3Config.getSecretKey())
-                .withApiCallTimeout(s3Config.getApiCallTimeout())
-                .forcePathStyleAccess(s3Config.isForcePathStyleAccess())
-                .withCompressionAlgorithm(s3Config.getCompression())
-                .withBasePath(s3Config.getBasePath())
-                .build();
-        S3BackupStore.validateConfig(storeConfig);
-      } catch (final Exception e) {
-        throw new InvalidConfigurationException("Cannot configure S3 backup store.", e);
+    try {
+      switch (backup.getStore()) {
+        case NONE -> LOG.warn("No backup store is configured. Backups will not be taken");
+        case S3 -> S3BackupStore.validateConfig(S3BackupStoreConfig.toStoreConfig(backup.getS3()));
+        case GCS -> GcsBackupStore.validateConfig(
+            GcsBackupStoreConfig.toStoreConfig(backup.getGcs()));
+        default -> throw new UnsupportedOperationException(
+            "Does not support validating configuration of backup store %s"
+                .formatted(backup.getStore()));
```

Comment:
:wrench: would be great to move these validations into the init methods of the configs, similar how we did it in https://github.com/camunda/zeebe/pull/11568/files

Reply:
I applied your suggestion at first but I'd like to revert this. Otherwise we can't easily unit test configuration anymore because simply using `TestConfigReader` will attempt to connect to GCS.